pennylane/pulse/rydberg.py

```diff
@@ -114,6 +114,147 @@ def circuit():
+    describing drive between the ground and excited states. The third term corresponds to the shift term
+    due to detuning from resonance. This drive term corresponds to a global drive that acts on all 4 wires of
+    the device.
+
+    The full Hamiltonian can be evaluated:
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit.jax", wires=wires)
+        @qml.qnode(dev, interface="jax")
+        def circuit(params):
+            qml.evolve(H_i + H_d)(params, t=[0, 10])
+            return qml.expval(qml.PauliZ(0))
+
+    >>> params = [2.4]
+    >>> circuit(params)
+    Array(0.94301294, dtype=float64)
+    >>> jax.grad(circuit)(params)
+    [Array(0.59484969, dtype=float64)]
+
+    We can also create a Hamiltonian with multiple local drives. The following circuit corresponds to the
+    evolution where an additional local drive acting on wires ``[0, 1]`` is added to the Hamiltonian:
+
+    .. code-block:: python
+
+        amplitude_local = lambda p, t: p[0] * jnp.sin(2 * jnp.pi * t) + p[1]
+        detuning_local = lambda p, t: p * jnp.exp(-0.25 * t)
+        phase_local = jnp.pi / 4
+        H_local = qml.pulse.rydberg_drive(amplitude_local, detuning_local, phase_local, [0, 1])
+
+        H = H_i + H_d + H_local
+
+        @jax.jit
+        @qml.qnode(dev, interface="jax")
+        def circuit_local(params):
+            qml.evolve(H)(params, t=[0, 10])
+            return qml.expval(qml.PauliZ(0))
+
+    >>> params = [2.4, [1.3, -2.0], -1.5]
+    >>> circuit_local(params)
+    Array(0.80238028, dtype=float64)
+    >>> jax.grad(circuit_local)(params)
+    [Array(-0.73273312, dtype=float64),
+     [Array(-0.45635171, dtype=float64, weak_type=True),
+      Array(0.76572817, dtype=float64, weak_type=True)],
+     Array(0.01027312, dtype=float64)]
+    """
+    if isinstance(wires, int):
+        wires = [wires]
+    trivial_detuning = not callable(detuning) and qml.math.isclose(detuning, 0.0)
+
+    if not callable(amplitude) and qml.math.isclose(amplitude, 0.0):
+        if trivial_detuning:
+            raise ValueError(
+                "Expected non-zero value for at least one of either amplitude or detuning, but "
+                f"received amplitude={amplitude} and detuning={detuning}. All terms are zero."
+            )
+
+        amplitude_term = None
+
+    else:
+        amplitude_term = drive(amplitude, phase, wires)
+
+    detuning_term = None
+
+    if not trivial_detuning:
+        detuning_obs = [-1.0 * sum(qml.PauliZ(wire) for wire in wires)]
+        detuning_coeffs = [detuning]
+        detuning_term = HardwareHamiltonian(detuning_coeffs, detuning_obs)
+
+    pulses = [HardwarePulse(amplitude, phase, detuning, wires)]
+
+    if amplitude_term and detuning_term:
+        drive_term = amplitude_term + detuning_term
+    elif amplitude_term:
+        drive_term = amplitude_term
+    else:
+        drive_term = detuning_term
+
+    drive_term.pulses = pulses
```

Comment:
100% agree that `HardwarePulse` creation be removed from `drive` and only be kept here, no reason for it to be in `drive`